File: leet/0094-binary-tree-inorder-traversal/binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    if(root == nil) {return []int{}}
    
    rs := make([]int, 0)
    rs = append(rs, inorderTraversal(root.Left)...)
    rs = append(rs, root.Val)
    rs = append(rs, inorderTraversal(root.Right)...)
    
    return rs
}

File: leet/0094-binary-tree-inorder-traversal/binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    if (!root) return []
    
    let cur = root
    let stack = []
    let ret = []
    
    while (cur){
        stack.push(cur)
        cur = cur.left
    }
    
    while(stack.length > 0){
        cur = stack.pop()
        ret.push(cur.val)
        
        if(cur.right){
            cur = cur.right
            while(cur){
                stack.push(cur)
                cur = cur.left
            }
        }
    }
    
    return ret
};
